infrastructure/src/main/java/org/corfudb/infrastructure/FailureHandlerDispatcher.java

```diff
@@ -205,7 +238,10 @@ private void reconfigureSequencerServers(CorfuRuntime runtime, Layout originalLa
                 verifyStreamTailsMap(streamTails);
 
                 // Configuring the new sequencer.
-                newLayout.getSequencer(0).reset(maxTokenRequested + 1, streamTails).get();
```

Comment:
the stream-tails discovery above is not safe. 

to make it safe, we must not allow progress unless we obtain the log tail from a read-quorum in every segment. A read-quorum is defined by the replication view (in chain, it is any single replica, for example). We can use the Quorum utils to perform the predicate-counting for us.

another alternative is to utilize the fastSmrMapsLoader to discover the tail. it might be simpler?